add-on/chrome/test/mochitest/browser_mozLoop_sharingListeners.js

```diff
@@ -49,15 +49,12 @@ function* promiseWindowIdReceivedNewTab(handlersParam = []) {
 }
 
 function promiseNewTabLocation() {
-  return new Promise(resolve => {
-    BrowserOpenTab();
-    let tab = gBrowser.selectedTab;
-    createdTabs.push(tab);
+  BrowserOpenTab();
+  let tab = gBrowser.selectedTab;
+  createdTabs.push(tab);
 
-    tab.linkedBrowser.addEventListener("DOMContentLoaded", function() {
-      resolve(tab.linkedBrowser.currentURI.spec);
-    });
-  });
+  // Have the tab's content process pass back its location as a promise
+  return ContentTask.spawn(tab.linkedBrowser, "", () => content.location.href);
```

Comment:
s/""/null/ is also a good value. Perhaps it might be good to allow 2 arguments to be passed to `ContentTask.spawn`. Oh well, perhaps I'll get to that someday!